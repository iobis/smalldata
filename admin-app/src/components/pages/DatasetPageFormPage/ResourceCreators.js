import React from 'react'
import ContactsForm from './ContactsForm/ContactsForm'
import { useTranslation } from 'react-i18next'

export default function ResourceCreators(props) {
  const { t } = useTranslation()

  return (
    <ContactsForm
      className="resource-creators"
      contactsTableHeader={t('datasetPageFormPage.resourceCreators.contactsTableHeader')}
      {...props}/>
  )
}
