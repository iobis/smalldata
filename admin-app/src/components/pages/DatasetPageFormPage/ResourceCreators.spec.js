import ResourceCreators from './ResourceCreators'
import React from 'react'
import { mount } from 'enzyme'

describe('ResourceCreators', () => {
  it('renders correctly', () => {
    const wrapper = mount(
      <ResourceCreators
        data={[]}
        onChange={jest.fn()}/>)
    expect(wrapper).toMatchSnapshot()
  })
})
