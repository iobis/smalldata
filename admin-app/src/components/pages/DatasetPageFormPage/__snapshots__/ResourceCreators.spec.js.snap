// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourceCreators renders correctly 1`] = `
<ResourceCreators
  data={Array []}
  onChange={[MockFunction]}
>
  <ContactsForm
    className="resource-creators"
    contactsTableHeader="datasetPageFormPage.resourceCreators.contactsTableHeader"
    data={Array []}
    onChange={[MockFunction]}
  >
    <div
      className="contacts-form resource-creators"
    >
      <OceanExpertNameInput
        oceanExpertName=""
        onChange={[Function]}
      >
        <div
          className="ocean-expert-name-input field column is-four-fifths mandatory"
        >
          <div
            className="dropdown"
          >
            <div
              className="dropdown-trigger"
            >
              <label
                className="label"
              >
                userFormPage.name.label
              </label>
              <div
                className="control has-icons-right "
              >
                <input
                  className="input"
                  onChange={[Function]}
                  onClick={[Function]}
                  placeholder=""
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              className="dropdown-menu"
              id="dropdown-menu"
              role="menu"
            >
              <div
                className="dropdown-content"
              />
            </div>
          </div>
        </div>
      </OceanExpertNameInput>
      <div
        className="columns column"
      >
        <InputText
          className="column email"
          name="datasetPageFormPage.contactsForm.email"
          onChange={[Function]}
          value=""
        >
          <div
            className="column field column email"
          >
            <label
              className="label"
              htmlFor="datasetpageformpage.contactsform.email.label"
            >
              datasetPageFormPage.contactsForm.email.label
            </label>
            <input
              className="input"
              id="datasetpageformpage.contactsform.email.label"
              onChange={[Function]}
              type="text"
              value=""
            />
          </div>
        </InputText>
        <InputText
          className="column organisation"
          name="datasetPageFormPage.contactsForm.organisation"
          onChange={[Function]}
          value=""
        >
          <div
            className="column field column organisation"
          >
            <label
              className="label"
              htmlFor="datasetpageformpage.contactsform.organisation.label"
            >
              datasetPageFormPage.contactsForm.organisation.label
            </label>
            <input
              className="input"
              id="datasetpageformpage.contactsform.organisation.label"
              onChange={[Function]}
              type="text"
              value=""
            />
          </div>
        </InputText>
        <div
          className="column field button-placeholder"
        >
          <button
            className="add button"
            onClick={[Function]}
          >
            common.add
          </button>
        </div>
      </div>
      <div
        className="column field"
      >
        <div
          className="title is-5 contacts-table-header"
        >
          datasetPageFormPage.resourceCreators.contactsTableHeader
        </div>
        <table
          className="general table is-fullwidth is-striped is-hoverable"
        >
          <thead>
            <tr>
              <th>
                datasetPageFormPage.contactsForm.name.label
              </th>
              <th>
                datasetPageFormPage.contactsForm.email.label
              </th>
              <th>
                datasetPageFormPage.contactsForm.organisation.label
              </th>
              <th />
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
  </ContactsForm>
</ResourceCreators>
`;
